tests/forms_tests/tests/test_forms.py

```diff
@@ -1369,6 +1369,22 @@ class UserRegistration(Form):
         self.assertEqual(bound['password'].value(), 'foo')
         self.assertEqual(unbound['password'].value(), None)
 
+    def test_boundfield_initial_called_once(self):
+        # Ensures multiple calls to BoundField().value() in an unbound form
+        # only results in a single call to a provided initial callable (#24391)
+        initial = mock.Mock(return_value='dynamic')
+
+        class Element(Form):
+            name = CharField(max_length=10, initial=initial)
```

Comment:
I'm in favor of avoiding mock where possible, so I would use something like `initial=uuid.uuid4` and simply verify the values returned in the two `value()` calls are equal. Are you okay with that approach?